src/mumble/GlobalShortcut_xdp.cpp

```diff
@@ -0,0 +1,189 @@
+    }
+
+    QDBusArgument arg;
+    arg << initialShortcuts;
+    auto reply = s_shortcutsInterface->CreateSession({
+        { QLatin1String("session_handle_token"), "Mumble" },
+        { QLatin1String("handle_token"), QLatin1String("mumble") },
+        { QLatin1String("shortcuts"), QVariant::fromValue(arg) },
+    });
+    reply.waitForFinished();
+    if (reply.isError()) {
+        qWarning() << "Couldn't get reply";
+        qWarning() << "Error:" << reply.error().message();
+    } else {
+        QDBusConnection::sessionBus().connect(QString(),
+                                            reply.value().path(),
+                                            QLatin1String("org.freedesktop.portal.Request"),
+                                            QLatin1String("Response"),
+                                            this,
+                                            SLOT(gotGlobalShortcutsCreateSessionResponse(uint,QVariantMap)));
+    }
+}
+
+void GlobalShortcutXdp::run() {
+	// 🤘🤪
+}
+
+bool GlobalShortcutXdp::canDisable() {
+	return false;
+}
+
+GlobalShortcutXdp::ButtonInfo GlobalShortcutXdp::buttonInfo(const QVariant &v) {
+	ButtonInfo info;
+	bool ok;
+	unsigned int key = v.toUInt(&ok);
+	if (!ok) {
+		return info;
+	}
+
+    info.device       = tr("Desktop");
+    info.devicePrefix = QString();
+    const auto id = m_ids[key];
+    for (const auto &x : m_shortcuts) {
+        if (x.first == id) {
+            info.name = x.second["trigger_description"].toString();
+        }
+    }
+	return info;
+}
+
+static QString parentWindowId()
+{
+    if (qEnvironmentVariableIsSet("WAYLAND_DISPLAY")) {
+        // TODO
+        return {};
+    }
+    return QLatin1String("x11:") + QString::number(qApp->focusWindow()->winId());
+}
+
+void GlobalShortcutXdp::gotGlobalShortcutsCreateSessionResponse(uint code, const QVariantMap &results)
+{
+    if (code != 0) {
+        qWarning() << "failed to create a global shortcuts session" << code << results;
+        return;
+    }
+
+    m_globalShortcutsSession = QDBusObjectPath(results["session_handle"].toString());
+
+    auto reply = s_shortcutsInterface->ListShortcuts(m_globalShortcutsSession, {});
+    reply.waitForFinished();
+    if (reply.isError()) {
+        qWarning() << "failed to call ListShortcuts" << reply.error();
+        return;
+    }
+
+    auto req = new OrgFreedesktopPortalRequestInterface(QLatin1String("org.freedesktop.portal.Desktop"),
+                                                        reply.value().path(), QDBusConnection::sessionBus(), this);
+
+    // BindShortcuts and ListShortcuts answer the same
+    connect(req, &OrgFreedesktopPortalRequestInterface::Response, this, &GlobalShortcutXdp::gotListShortcutsResponse);
+    connect(req, &OrgFreedesktopPortalRequestInterface::Response, req, &QObject::deleteLater);
+}
+
+void GlobalShortcutXdp::gotListShortcutsResponse(uint, const QVariantMap &results)
+{
+    const auto arg = results["shortcuts"].value<QDBusArgument>();
```

Comment:
If `arg` is of type `QDbusArgument`, then using `auto` here is fine. Otherwise, auto -> actual type

Reply:
I just switched all autos to the type.